python/lsst/ts/tcpip/base_one_client_read_loop_server_test_case.py

```diff
@@ -0,0 +1,58 @@
+# This file is part of ts_tcpip.
+#
+# Developed for the Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import abc
+import asyncio
+import contextlib
+import logging
+import unittest
+from typing import AsyncGenerator
+
+from .client import Client
+from .one_client_read_loop_server import OneClientReadLoopServer
+
+__all__ = ["BaseOneClientReadLoopServerTestCase"]
+
+# Standard timeout in seconds.
+TIMEOUT = 2
+
+
+class BaseOneClientReadLoopServerTestCase(unittest.IsolatedAsyncioTestCase):
+    async def asyncSetUp(self) -> None:
+        self.log = logging.getLogger(type(self).__name__)
+
+    @contextlib.asynccontextmanager
+    async def create_server_and_client(self) -> AsyncGenerator[None, None]:
+        self.srv = await self.create_server()
+        await self.srv.start_task
```

Comment:
Please add code that reliably shuts down the client and server if the test fails.

For the client, creating it as an async context manager is the simplest solution I know of.

You could do the same for the server by making create_server an async context manager, but it complicates the make_server method slightly. The obvious alternative is try/finally.

I also request that you name the attribute server, not srv. We try to avoid abbreviations where practical. It does lead to the awkward self.server.server, but I do not expect users to ever mess with the server attribute of a OneClientServer, so in practice I think it's fine.

I probably should have made server a private attribute of OneClientServer. Feel free to change it to that by adding an underscore. I'm a bit uneasy about accessing it here in the test. Have you found that it catches anything?
```
self.server = await self.create_server()
try:
    await self.server.start_task
    async with Client(...) as self.client:
        yield
finally:
    await self.server.close()
```

Reply:
Thanks for catching this. Fixed.